dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java

```diff
@@ -125,6 +171,14 @@ public void addSyncHandler(SynchronousHandler syncHandler) {
 		 */
 		RERWLOCK.writeLock().lock();
 		try {
+			//Send messages from EMQ to synchronous handler when it subscribes to receive messages
+			if(earlyMessageQueue != null && !earlyMessageQueue.isEmpty() && !synchronizedHandlerSet.contains(syncHandler)) {
+				Object holder[] = new Object[EARLY_MESSAGE_QUEUE_SIZE];
+				Object[] messagesList = earlyMessageQueue.toArray(holder);
```

Comment:
it doesn't make sense to copy the EMQ here since you need to avoid anyone adding to the EMQ while you're doing the copy to the syncHandler. You can just copy straight from the EMQ into the syncHandler.